app/src/main/java/com/sun/noteapp/ui/home/NoteVerticalWideAdapter.kt

```diff
@@ -0,0 +1,120 @@
+import kotlinx.android.synthetic.main.item_note_vertical_wide.view.*
+
+class NoteVerticalWideAdapter(
+    private val listener: OnNoteItemClick,
+    private val maxWidthItem: Int
+) : BaseRecyclerViewAdapter<Note, NoteVerticalWideAdapter.ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(
+            R.layout.item_note_vertical_wide,
+            parent,
+            false
+        )
+        return ViewHolder(view, listener)
+    }
+
+    inner class ViewHolder(
+        itemView: View,
+        private val listener: OnNoteItemClick
+    ) : BaseViewHolder<Note>(itemView) {
+
+        override fun onBindData(itemData: Note) {
+            super.onBindData(itemData)
+            itemView.apply {
+                textTitleNoteVerticalWide.setBackgroundResource(
+                    ColorPicker.getMediumColor(itemData.color)
+                )
+                layoutContentNoteVerticalWide.setBackgroundResource(
+                    ColorPicker.getLightColor(itemData.color)
+                )
+                textTitleNoteVerticalWide.text = itemData.title
+                val content = if (itemData.type == 1) itemData.content
+                else ConvertString.showActionCheckListByDataString(itemData.content)
+                textContentNoteVerticalWide.text = content
+                if (itemData.remindTime == NONE)
+                    textRemindTimeNoteVerticalWide.visibility = View.GONE
+                else {
+                    textRemindTimeNoteVerticalWide.visibility = View.VISIBLE
+                    textRemindTimeNoteVerticalWide.text = itemData.remindTime
+                }
+                if (itemData.label != NONE) {
+                    layoutLabelNoteVerticalWide.removeAllViewsInLayout()
+                    addLabelView(itemData.label, itemView)
+                } else {
+                    layoutLabelNoteVerticalWide.visibility = View.GONE
+                }
+                if (itemData.password != NONE) {
+                    iconLockNoteVerticalWide.visibility = View.VISIBLE
+                    layoutLabelNoteVerticalWide.visibility = View.GONE
+                    textContentNoteVerticalWide.text = EMPTY
+                } else {
+                    iconLockNoteVerticalWide.visibility = View.GONE
+                }
+            }
+        }
+
+        override fun onHandleItemClick(mainItem: Note) {
+            super.onHandleItemClick(mainItem)
+            listener.showNoteDetail(adapterPosition, mainItem)
+        }
+
+    }
+
+    @SuppressLint("SetTextI18n")
+    fun addLabelView(label: String, view: View) {
+        view.apply {
+            val labels = ConvertString.labelStringDataToLabelList(label)
+            var sum = 0
+            for (i in labels.indices) {
+                val tv = TextView(context)
+                val params = LinearLayout.LayoutParams(
```

Comment:
mỗi một label lại phải tạo lại params sao?

Reply:
nếu dùng chung a add label vào layout parent nó bị lỗi, kiểu bị trùng hay s ý
The specified child already has a parent. You must call removeView() on the child's parent first.